lib/ref-issue-plugin.js

```diff
@@ -0,0 +1,57 @@
+module.exports = (robot) => {
+    robot.on('pull_request.opened', async context => {
+        const pull_request = context.payload.pull_request
+        var title = pull_request.title
+        title = title.toString()
+        title = title.toLowerCase()
+        title = title.toString().split(" ").join("")
+        console.log(title)
+        if(title.slice(0,3) == 'dev'){
+            string_index=4;
+            while(string_index<title.length && title[string_index] != '#'){
+                string_index += 1;
+            }
+            var link_head = title.slice(4,string_index)
+            if(string_index == title.length){
+                return;
+            }
+            var issue_number_index = string_index+1;
+            while(issue_number_index<title.length && title[issue_number_index]-'0'>=0 && title[issue_number_index]-'0'<=9){
+                issue_number_index += 1
+            }
+            var link_number = title.slice(string_index+1,issue_number_index)
+            var ref_link = 'https://lab.civicrm.org/dev' + '/' + link_head + '/-/issues/' + link_number
+            ref_link = 'The issue assosciated with the Pull Request can be viewed on this link ' + ref_link
+
+            var project_ids = require('./project_id.js');
+            const { Gitlab } = require('@gitbeaker/node');
+            const api = new Gitlab({
+                host: 'https://lab.civicrm.org',
+                token: 'jjXgPeCPuRRSqHkDdy4T',
```

Comment:
Is this a real token for CiviCRM's GitLab?

Reply:
Secrets should always be set with environment variables to prevent things like this!

Reply:
Yes, I am working on removing it from here and directly adding it as .env variable from where I am hosting on it. I did not know at first on how to pass ..env variables and was working on it now only.

Reply:
Right, but you've leaked a live token with access to GitLab projects on the public internet.

Reply:
I am really sorry, I was not aware if that could be used in some wrong way, I have removed it as of now, If its of any use. I am really sorry, I will take care of it in future :(